velox/dwio/parquet/tests/reader/ParquetReaderBenchmark.cpp

```diff
@@ -0,0 +1,211 @@
+        bool hasData = rowReader->next(nextSize, result);
+        if (!hasData) {
+          break;
+        }
+
+        if (result->size() == 0) {
+          continue;
+        }
+
+        auto rowVector = result->asUnchecked<RowVector>();
+        for (auto i = 0; i < rowVector->childrenSize(); ++i) {
+          rowVector->childAt(i)->loadedVector();
+        }
+      }
+    }
+
+    rowReader->updateRuntimeStats(runtimeStats_);
+  }
+
+  void readSingleColumnPlain(
+      const ParquetReaderType& parquetReaderType,
+      const std::string columnName,
+      const TypePtr& type,
+      uint8_t nullsRateX100,
+      const FilterSpec& filterSpec,
+      uint32_t nextSize) {
+    folly::BenchmarkSuspender suspender;
+
+    auto rowType = ROW({columnName}, {type});
+    auto batches =
+        dataSetBuilder_->makeDataset(rowType, kNumBatches, kNumRowsPerBatch)
+            .withRowGroupSpecificData(kNumRowsPerRowGroup)
+            .withNullsForField(Subfield(columnName), nullsRateX100)
+            .build();
+    writeToFile(*batches, true);
+
+    auto scanSpec = createScanSpec(*batches, rowType, {filterSpec});
+
+    suspender.dismiss();
+
+    read(parquetReaderType, rowType, scanSpec, nextSize);
+  }
+
+ private:
+  std::unique_ptr<test::DataSetBuilder> dataSetBuilder_;
+  std::unique_ptr<memory::MemoryPool> pool_;
+  dwio::common::DataSink* sinkPtr_;
+  std::unique_ptr<facebook::velox::parquet::Writer> writer_;
+  RuntimeStatistics runtimeStats_;
+};
+
+BENCHMARK(single_column_bigint_plain_nonull_filterNothing_10000) {
+  std::string columnName = "bigint";
+  FilterSpec filterSpec(columnName, 0, 100, FilterKind::kBigintRange, true);
+
+  ParquetReaderBenchmark benchmark;
+  benchmark.readSingleColumnPlain(
+      ParquetReaderType::NATIVE, columnName, BIGINT(), 0, filterSpec, 10000);
+}
+
+BENCHMARK(single_column_bigint_plain_partialnulls_filterNothing_10000) {
+  std::string columnName = "bigint";
+  FilterSpec filterSpec(columnName, 0, 100, FilterKind::kBigintRange, true);
+
+  ParquetReaderBenchmark benchmark;
+  benchmark.readSingleColumnPlain(
+      ParquetReaderType::NATIVE, columnName, BIGINT(), 50, filterSpec, 10000);
+}
+
+BENCHMARK(single_column_bigint_plain_nonull_filter50_10000) {
+  std::string columnName = "bigint";
+  FilterSpec filterSpec(columnName, 0, 50, FilterKind::kBigintRange, true);
+
+  ParquetReaderBenchmark benchmark;
+  benchmark.readSingleColumnPlain(
+      ParquetReaderType::NATIVE, columnName, BIGINT(), 0, filterSpec, 10000);
+}
+
+BENCHMARK(single_column_bigint_plain_partialnulls_filter50_10000) {
+  std::string columnName = "bigint";
+  FilterSpec filterSpec(columnName, 0, 50, FilterKind::kBigintRange, true);
+
+  ParquetReaderBenchmark benchmark;
+  benchmark.readSingleColumnPlain(
+      ParquetReaderType::NATIVE, columnName, BIGINT(), 50, filterSpec, 10000);
+}
+
+// TODO: Add all data types
+// TODO: Add dictionary encoded data
+
+int main(int /*argc*/, char** /*argv*/) {
+  folly::runBenchmarks();
```

Comment:
Call `folly::init` here and you can get a few command line options controlling the benchmark, i.e. `--bm_regex`